### DISCUSSION TITLE
How to Upgrade from Coolify v3 to v4

### DISCUSSION THREAD

**Discussion by ducphuclee** [coollabsio/coolify#1617] [Q&A]
I am currently using Coolify v3 and would like to know how to upgrade to version 4. I have searched through the documentation and various online forums but could not find specific instructions.

Are there any guides available that detail the steps to upgrade from Coolify v3 to v4? What steps should I take to ensure that all my data and configurations remain intact during the upgrade?

I look forward to your support.

Thank you very much!

**Comment by cruonline** [comment]
anyone? @andrasbacsai can you give a short answer to this? Is it possible or is a clean installation needed?

**Comment by vermaxik** [comment]
I found this thread https://github.com/coollabsio/coolify/discussions/1027, might be helpful

>Q: Is v3 upgradable to v4?
A: Not in a straightforward way (just by clicking on the update button on Coolify’s UI), but I will add an option to import your resources from v3 to a v4 instance.I try to make it the most simple and easy task, without any downtime. I will keep you updated! There will be a migration path available.

@andrasbacsai is it somewhere guide how to migrate? Thank you in advance 🙏

**Comment by preslavrachev** [comment]
I am also currently stuck in this situation. My instance runs on the last available v3 release (3.12.37). While it keeps my existing apps running, I noticed it getting behind on certain things. So, I'd also love to be able to migrate to v4. Even pay for that if i have to.

@andrasbacsai any easy way to migrate?
